minio/api.py

```diff
@@ -193,6 +196,14 @@ def make_bucket(self, bucket_name, location='us-east-1'):
         """
         is_valid_bucket_name(bucket_name)
 
+        ## Region already set in constructor, validate if
+        ## caller requested bucket location is same.
+        if self._region:
+            if self._region != location:
+                raise InvalidArgumentError("Configured region {0}, requested"
```

Comment:
Alternatively, instead of doing something that appears to change behaviour a lot, we can get equivalent behaviour by modifying the pre-sign APIs to take an explicit region argument.